pkg/segment/set.go

```diff
@@ -932,11 +1001,150 @@ func (n *node) rebalanceAfterRemove(gap GapIterator) GapIterator {
 		}
 		p.children[p.nrSegments] = nil
 		p.nrSegments--
+		// update maxGap of left, no need to change p and right
+		// right is already invalid
+		left.updateLocalMaxGap()
 		// This process robs p of one segment, so recurse into rebalancing p.
 		n = p
 	}
 }
 
+// update maxGap bottom up from the calling leaf until no updates
```

Comment:
All function comment should conform to godoc standards. E.g.

updateMaxGap bottoms up ...

Reply:
(Here and elsewhere.)